contrib_bots/lib/john.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+import time
+import sys
+
+from random import choice
+
+try:
+    from chatterbot import ChatBot
+    from chatterbot.trainers import ChatterBotCorpusTrainer, ListTrainer
+except ImportError:
+    raise ImportError("""It looks like you are missing chatterbot.
+                      Please: pip install chatterbot""")
+
+CONTRIB_BOTS_DIR = os.path.dirname(os.path.abspath('__filename__'))
```

Comment:
It was working... But I realized any filename will do in place of "__file__". E.g. realpath('nonexistentfile.whatever'). I'm getting an error without the quotes: 
`Traceback (most recent call last):
  File "lib/john.py", line 15, in <module>
    CONTRIB_BOTS_DIR = os.path.dirname(os.path.abspath(__filename__))
NameError: name '__filename__' is not defined
(zulip-venv) ~/local/zulip/contrib_bots (joke)`